app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/Search.php

```diff
@@ -141,4 +150,26 @@ private function buildSearchCriteria(array $args, ResolveInfo $info): SearchCrit
 
         return $searchCriteria;
     }
+
+
+    /**
+     * Get product attribute value
+     *
+     * @param Product $product
+     * @param $productArray
+     * @param $field
+     *
+     * @return string|null
+     */
+    private function getAttributeValue(Product $product, $productArray, $field)
+    {
+        if ($attribute = $product->getResource()->getAttribute($field)) {
+            $attributeValue =  $attribute->getFrontend()->getValue($product);
+            if ($attributeValue && !($attributeValue instanceof Phrase) ) {
+               return $attributeValue;
+            } else {
+                return isset($productArray[$product->getId()][$field])? $productArray[$product->getId()][$field] : null;
```

Comment:
We may simplify this structure with null coalescing operator.
```suggestion
                return $productArray[$product->getId()][$field] ?? null;
```

Reply:
If we will use simplify condition we can get error "Undefined index" in case where product don't have requested attribute. `isset()` helps to avoid this

Reply:
The null coalescing operator should not generate the "Undefined Index" notice. Please, double-check it.

Reply:
Hi @rogyar  , Yes you are right .
I have applied this changes 
Thanks